clients/java/zpe/src/main/java/com/yahoo/athenz/zpe/AuthZpeClient.java

```diff
@@ -165,6 +210,68 @@ public static PublicKey getZmsPublicKey(String keyId) {
     
     /**
      * Determine if access(action) is allowed against the specified resource by
+     * a user represented by the X509Certificate
+     * @param cert - X509Certificate
+     *        
+     * @param angResource is a domain qualified resource the calling service
+     *        will check access for.  ex: my_domain:my_resource
+     *        ex: "angler:pondsKernCounty"
+     *        ex: "sports:service.storage.tenant.Activator.ActionMap"
+     * @param action is the type of access attempted by a client
+     *        ex: "read"
+     *        ex: "scan"
+     * @return AccessCheckStatus if the user can access the resource via the specified action
+     *        the result is ALLOW otherwise one of the DENY_* values specifies the exact
+     *        reason why the access was denied
+     */
+    public static AccessCheckStatus allowAccess(X509Certificate cert, String angResource, String action) {
+        StringBuilder matchRoleName = new StringBuilder(256);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("AUTHZPECLT:allowAccess: action=" + action + " resource=" + angResource);
+        }
+        zpeMetric.increment(ZpeConsts.ZPE_METRIC_NAME, DEFAULT_DOMAIN);
+        // validate the certificate against CAs
+        X500Principal issuerx500Principal = cert.getIssuerX500Principal();
+        String issuer = issuerx500Principal.getName();
+        
+        if (issuer == null || issuer.isEmpty() 
+                || !X509_ISSUERS.contains(issuer.replaceAll("\\s+" , ""))) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("AUTHZPECLT:allowAccess: missing or mismatch issuer {}", issuer);
+            }
+            return AccessCheckStatus.DENY_CERT_MISMATCH_ISSUER;
+        }
+        String subject = Crypto.extractX509CertCommonName(cert);
+        if (subject == null || subject.isEmpty()) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("AUTHZPECLT:allowAccess: missing subject");
+            }
+            return AccessCheckStatus.DENY_CERT_MISSING_SUBJECT;
+        }
+        int idx = subject.indexOf(":");
+        if (idx == -1) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("AUTHZPECLT:allowAccess: missing domain");
+            }
+            return AccessCheckStatus.DENY_CERT_MISSING_DOMAIN;
+        }
+        String domainName = subject.substring(0, idx);
+        int roleIndex = subject.indexOf(".");
```

Comment:
This check is not correct since the domain name can contain a "."
e.g. sys.auth:role.readers
we should search for ":role." for the domain name/role name index. Anything before that is the domain name and everything after is the role name